xs/src/libslic3r/MedialAxis.cpp

```diff
@@ -1005,18 +1005,8 @@ MedialAxis::main_fusion(ThickPolylines& pp)
                 break;
             }
         }
-        if (changes) {
-            concatThickPolylines(pp);
-            ///reorder, in case of change
-            std::sort(pp.begin(), pp.end(), [](const ThickPolyline & a, const ThickPolyline & b) {
-                bool ahas0 = a.width.front() == 0 || a.width.back() == 0;
-                bool bhas0 = b.width.front() == 0 || b.width.back() == 0;
-                if (ahas0 && !bhas0) return true;
-                if (!ahas0 && bhas0) return false;
-                return a.length() < b.length();
-            });
-        }
     }
+    if (changes) concatThickPolylines(pp);
```

Comment:
loop is exited only when changes == false

Reply:
right